felix/bpf-gpl/parsing.h

```diff
@@ -129,11 +167,12 @@ static CALI_BPF_INLINE int tc_state_fill_from_nexthdr(struct cali_tc_ctx *ctx)
 	switch (ctx->state->ip_proto) {
 	case IPPROTO_TCP:
 		// Re-check buffer space for TCP (has larger headers than UDP).
-		if (skb_refresh_validate_ptrs(ctx, TCP_SIZE)) {
+		if (validate_ptrs(ctx, TCP_SIZE)) {
```

Comment:
This does not pull extra data if the packet is long enough but there is not enough for direct access. Perhaps that is the problem :thinking:

Reply:
Exactly, that's the issue I am trying to solve. If I use `skb_refresh_validate_ptrs` the verifier again complains and reject the program. It seems the issue is around `skb_refresh_start_end(ctx);`. But I have not figured it out yet.